mesonbuild/backend/ninjabackend.py

```diff
@@ -1808,7 +1808,7 @@ def generate_genlist_for_target(self, genlist, target, outfile):
                 elem.add_item('DEPFILE', depfile)
             if len(extra_dependencies) > 0:
                 elem.add_dep(extra_dependencies)
-            elem.add_item('DESC', 'Generating $out')
+            elem.add_item('DESC', 'Generating \'%s\'.' % sole_output)
```

Comment:
So then maybe the there is a bug by escaping special characters instead? The generated code looks like this:

```
build yasm_app@exe/yasm.asm.o: CUSTOM_COMMAND ../src/yasm.asm
 DESC = Generating$ $$out
 COMMAND = /usr/bin/yasm ../src/yasm.asm -o yasm_app@exe/yasm.asm.o -g dwarf2 -$
```